### DISCUSSION TITLE
Documentation required for Nuget Client SDK classes

### DISCUSSION THREAD

**Discussion by NavinAT** [NuGet/Home#12990] [Q&A] [answered]
Please provide documentation for 'https://github.com/NuGet/NuGet.Client/tree/dev/src' NuGet Client SDK classes. 
In Microsoft link 'https://learn.microsoft.com/en-us/nuget/what-is-nuget' couldn't find the class (definition, methods and properties) related information.

There are many classes in 'NuGet.Protocol dll' (more repository, providers & resources).

1. What are the uses of 'Repository, Providers, & Resources'?
2. How to work with local folder's and gallery's - Nupkg?
3. How to work with cache?
4. Some 'providers & resources - classes' have 'V2/V3': why do you have both?

Is there any proper documentation for this Nuget Client library.

**FYI:**
According to this link 'https://learn.microsoft.com/en-us/nuget/reference/nuget-client-sdk#third-party-documentation' - it covers only the basic information.

**Comment by nkolev92** [comment, marked as answer]
1. Nowadays, it's answered within the readme of NuGet.Protocol: https://www.nuget.org/packages/NuGet.Protocol#readme-body-tab. 
4. There are V2 and V3 http sources, as described here: https://learn.microsoft.com/en-us/nuget/api/overview#versioning. We need different implementations depending on the API version.
